### ISSUE TITLE
preprocess data

### ISSUE THREAD

**Issue by alpardayalman** [alpardayalman/Feature_engineering#7]
Once you have a good understanding of your data, it is time to get it prepared for modeling. These steps can involve joining data, standardizing variables, removing variables, imputation, and splitting data into train and test sets.

In order to answer the question you’ve formulated, you will probably need to create some new columns in the DataFrame. This is especially true because so much of our data here is categorical (i.e. diet consists of the options vegan, vegetarian, anything, etc. instead of numerical values).

Categorical data is great to use as labels, but we want to create some numerical data as well to use for features.

[labelEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html)
[standardization](https://scikit-learn.org/stable/modules/preprocessing.html)
[reshaping](https://pandas.pydata.org/pandas-docs/stable/user_guide/reshaping.html)
[merge and joining](https://pandas.pydata.org/pandas-docs/stable/user_guide/merging.html)
[splitting data](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.model_selection)
Main components that you will want to include:

Data ready for modeling
